Fix documentation workflow block of PRs and add names to jobs (#72)
eclipse-ankaios/ankaios: doc/docs/development/ci-cd.md

Example:

Bad:
```yaml
...
  - uses: actions/checkout@v3
...
```

Good:
```yaml
...
  - uses: actions/checkout@v3.5.3
...
```

## Adding Github action jobs
When creating a new job inside a workflow, specify a job name for each job.

Example:

```yaml
...

jobs:
  test_and_build_linux_amd64:
    name: Test and Build Linux amd64
...
```

!!! Note

>     Giving a job a name is a best practice and if a job needs to configured as required status check inside the [self-service repository](https://github.com/eclipse-ankaios/.eclipsefdn) then this job is referenced by its name inside the configuration.

Reviewer: ```suggestion
    Beside being a best practice, giving a job a name is needed to reference it from the [self-service repository](https://github.com/eclipse-ankaios/.eclipsefdn) in order to configure the job as a required status check.
```
